### PR TITLE
Update of disabled unit tests

### LINKED ISSUE #155: Revisit Disabled unit tests
There a few tests in the tyr-core module that are disabled. Look into why is that the case and if they can be run again now.

### PR DESCRIPTION
**Pull request by xvanick1**
Fixes #155

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,6 +4,23 @@
...
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>2.12.1</version>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.dataformat</groupId>
+            <artifactId>jackson-dataformat-yaml</artifactId>
+            <version>2.12.1</version>
+        </dependency>
```
Comment (xstefank) [inline pom.xml:22]: why did you move these to parent pom?

File: pom.xml [hunk 2]
```diff
@@ -4,6 +4,23 @@
     <groupId>org.jboss.tyr</groupId>
     <artifactId>tyr-parent</artifactId>
     <version>1.1.0-SNAPSHOT</version>
+    <dependencies>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-annotations</artifactId>
+            <version>2.12.1</version>
```
Comment (xstefank) [inline pom.xml:11]: is this needed?

File: tyr-core/src/test/java/org/jboss/tyr/check/ValidFormatFileTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+@QuarkusTest
+class ValidFormatFileTest {
+    @Inject
+    TemplateChecker templateChecker;
+
+    @Test
+    public void testConfigSetValidFormatFilePath() throws InvalidPayloadException {
+        //Optional<String> optionalValue = ConfigProvider.getConfig().getOptionalValue("tyr.template.format.url", String.class);
+        //System.out.println("tyr.template.format.url=" + optionalValue.orElse("isEmpty"));
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/check/ValidFormatFileTest.java:20]: commented out code

File: tyr-core/src/test/java/org/jboss/tyr/config/EmptyAdditionalResourcesPropertyTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        }
+    }
+
+    @Test
+    public void emptyAdditionalResourcesPropertyTest() throws InvalidPayloadException {
+        String result = templateChecker.checkPR(TestUtils.TEST_PAYLOAD);
+        whitelistProcessing.processPRComment(TestUtils.ISSUE_PAYLOAD);
+
+        Assertions.assertTrue(result.isEmpty());
+        Assertions.assertFalse(DummyAdditionalCommand.isTriggered());
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/EmptyAdditionalResourcesPropertyTest.java:41]: It's always good idea to provide some failure message, can you provide some please?

File: tyr-core/src/test/java/org/jboss/tyr/config/FormatUndefinedTest.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    public static class Profile implements QuarkusTestProfile {
+        @Override
+        public Map<String, String> getConfigOverrides() {
+            return Map.of("tyr.template.format.file", "");
+        }
+    }
+
+    @Test
+    public void testConfigUnsetFormatFileUrlAndFilePath() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> templateChecker.checkPR(TestUtils.TEST_PAYLOAD));
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/FormatUndefinedTest.java:29]: same for all asserts without messages

File: tyr-core/src/test/java/org/jboss/tyr/config/FormatUndefinedTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package org.jboss.tyr.config;
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/FormatUndefinedTest.java:1]: license headers should be in all source files

File: tyr-core/src/test/java/org/jboss/tyr/config/InvalidFormatFileTest.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 Red Hat, Inc, and individual contributors.
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/InvalidFormatFileTest.java:2]: it's 2021 right? :D

File: tyr-core/src/test/java/org/jboss/tyr/config/InvalidFormatFileTest.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    public static class Profile implements QuarkusTestProfile {
+        @Override
+        public Map<String, String> getConfigOverrides() {
+            return Map.of("tyr.template.format.file", "src/test/resources/yaml/wrongTemplateInvalidFormat.yaml");
+        }
+    }
+
+    @Test
+    public void testNullConfigParameter() {
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/InvalidFormatFileTest.java:43]: testInvalidFormatYaml

**Review by xstefank — CHANGES_REQUESTED** [review]
Looks good, only some formatting comments and error messages.

File: pom.xml (hunk 2 above)
Comment (xstefank) [reply pom.xml:11]: you didn't answer this.

File: tyr-core/pom.xml [hunk 9]
```diff
@@ -58,12 +58,7 @@
     <dependency>
       <groupId>com.fasterxml.jackson.dataformat</groupId>
       <artifactId>jackson-dataformat-yaml</artifactId>
-      <version>${version.com.fasterxml.jackson}</version>
-    </dependency>
-    <dependency>
-      <groupId>com.fasterxml.jackson.core</groupId>
-      <artifactId>jackson-databind</artifactId>
-      <version>${version.com.fasterxml.jackson}</version>
+      <version>2.12.1</version>
```
Comment (xstefank) [inline tyr-core/pom.xml:61]: don't inline versions please, use properties

File: tyr-core/pom.xml [hunk 10]
```diff
@@ -58,12 +58,7 @@
     <dependency>
       <groupId>com.fasterxml.jackson.dataformat</groupId>
       <artifactId>jackson-dataformat-yaml</artifactId>
-      <version>${version.com.fasterxml.jackson}</version>
-    </dependency>
-    <dependency>
-      <groupId>com.fasterxml.jackson.core</groupId>
-      <artifactId>jackson-databind</artifactId>
-      <version>${version.com.fasterxml.jackson}</version>
```
Comment (xstefank) [inline tyr-core/pom.xml:66]: where else is this needed? I see you moved it to the parent.

File: tyr-core/src/test/java/org/jboss/tyr/config/FormatUndefinedTest.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        public Map<String, String> getConfigOverrides() {
+            return Map.of("tyr.template.format.file", "");
+        }
+    }
+
+    @Test
+    public void testConfigUnsetFormatFileUrlAndFilePath() {
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> templateChecker.checkPR(TestUtils.TEST_PAYLOAD),
+                "Expected IllegalArgumentException to be thrown");
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/FormatUndefinedTest.java:46]: ... when format is not defined

File: tyr-core/src/test/java/org/jboss/tyr/config/InvalidFormatFileTest.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        public Map<String, String> getConfigOverrides() {
+            return Map.of("tyr.template.format.file", "src/test/resources/yaml/wrongTemplateInvalidFormat.yaml");
+        }
+    }
+
+    @Test
+    public void testInvalidFormatYaml() {
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> templateChecker.checkPR(TestUtils.TEST_PAYLOAD),
+                "Expected IllegalArgumentException to be thrown");
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/InvalidFormatFileTest.java:46]: ... when ...

File: tyr-core/src/test/java/org/jboss/tyr/config/InvalidPathAdditionalResourcesTest.java [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    @Inject
+    WhitelistProcessing whitelistProcessing;
+
+    @Inject
+    TemplateChecker templateChecker;
+
+    public static class Profile implements QuarkusTestProfile {
+        @Override
+        public Map<String, String> getConfigOverrides() {
+            return Map.of("tyr.additional-resources", "rc/test/resources/generated/non-existent-custom-resources.jar");
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/InvalidPathAdditionalResourcesTest.java:45]: I understand that this is non-existent path but I think it still should be to existing directory in which we know we don't have the file, so it should be src/test/... not rc/test

File: tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+                throw new RuntimeException(e);
+            }
+
+            return Map.of("tyr.template.format.url", yamlPath);
+        }
+    }
+
+    @Test
+    public void testNullConfigParameter() {
+        Assertions.assertDoesNotThrow(() -> templateChecker.checkPR(TestUtils.TEST_PAYLOAD), "Expected nothing to be thrown");
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java:56]: Why don't you check here rather that the correct result is returned?

**Review by xstefank — CHANGES_REQUESTED** [review]

File: tyr-core/src/test/java/org/jboss/tyr/config/InvalidPathAdditionalResourcesTest.java (hunk 13 above)
Comment (xvanick1) [reply tyr-core/src/test/java/org/jboss/tyr/config/InvalidPathAdditionalResourcesTest.java:45]: Just a typo ;)

File: tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java (hunk 14 above)
Comment (xvanick1) [reply tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java:56]: Used previous code from the test. I'm not sure if it's good to bother with the hard-typed path comparison.

File: tyr-core/pom.xml (hunk 10 above)
Comment (xvanick1) [reply tyr-core/pom.xml:66]: Nowhere, deleted.

File: tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java (hunk 14 above)
Comment (xstefank) [reply tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java:56]: It's better to validate that the process correctly run and not that inly no exception was thrown. You can copy asserts from other valid path tests.

File: pom.xml (hunk 2 above)
Comment (xvanick1) [reply pom.xml:11]: Removed.

File: tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java (hunk 14 above)
Comment (xvanick1) [reply tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java]: Thank you for the recommendation.

File: pom.xml [hunk 15]
```diff
@@ -4,6 +4,8 @@
     <groupId>org.jboss.tyr</groupId>
     <artifactId>tyr-parent</artifactId>
     <version>1.1.0-SNAPSHOT</version>
+    <dependencies>
+    </dependencies>
```
Comment (xstefank) [inline pom.xml:8]: empty tags are not needed ;)

File: tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+                System.out.println(yamlPath);
+            } catch (MalformedURLException e) {
+                throw new RuntimeException(e);
+            }
+            return Map.of("tyr.template.format.url", yamlPath);
+        }
+    }
+
+    @Test
+    public void testNullConfigParameter() throws InvalidPayloadException {
```
Comment (xstefank) [inline tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java:56]: I just noticed this now, but this is probably not a right name for this test?

**Review by xstefank — CHANGES_REQUESTED** [review]
Last one, only two small changes needed.

File: tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java (hunk 16 above)
Comment (xvanick1) [reply tyr-core/src/test/java/org/jboss/tyr/config/ValidFormatUrlTest.java]: Yes, it's not a right name for this test. Thanks for mention, it should be okay now.

**Review by xstefank — APPROVED** [review]
